src/reflect/scala/reflect/internal/tpe/TypeComparers.scala

```diff
@@ -528,7 +528,7 @@ trait TypeComparers {
         (rt2.parents forall (isSubType(tp1, _, depth))) &&
           (rt2.decls forall (specializesSym(tp1, _, depth)))
       case et2: ExistentialType =>
-        et2.withTypeVars(isSubType(tp1, _, depth), depth) || fourthTry
```

Comment:
This works, but I'd prefer to keep the depth bookkeeping logic in GlbLubs. How about adding a `.decr` in this spot:
https://github.com/scala/scala/blob/a2cd2573ee9ef2cc856b92efa646b86524eea495/src/reflect/scala/reflect/internal/tpe/GlbLubs.scala#L354

In my superficial local testing, this also seems to break the cycle for this test case. What do you think?

Reply:
Ah yes, you are right, that works too! I wonder if this check is even needed though. The comment about higher-order type parameters shouldn't be true anymore.

Reply:
Yes! Definitely worth looking into getting rid of this additional check.